36-vending-machine-monitor/README.md

```diff
@@ -0,0 +1,369 @@
+* Low battery alerts
+* Sensor offline alerts for each column
+* Vending notifications for each column
+* Environment variable change notifications
+
+### Alerts
+
+Alerts are sent to the outbound queue `alert.qo`. An alert is sent when an alert condition becomes active, and once again when the alert condition is cleared.
+
+All alerts have these properties in common:
+
+* `active`: `true` or `false` to indicate whether the alert condition has just occurred or has been cleared.
+* `text`: A short description of the state of the alert.
+
+In addition to these properties, the Sensor Offline Alert also includes a `col` property with the column number that has gone offline.
+
+#### Tilt Alerts
+
+The app sends a tilt alert when vending machine is tilted past the angle given by the environment variable `tilt_alert_deg`.
+
+```json
+{ "active": true, "text": "excessive tilt" }
+```
+
+When the vending machine tilt returns to normal, the alert is cleared with:
+
+```json
+{ "active": false, "text": "no tilt" }
+```
+
+Tilt is detected using accelerometer readings from the Notecard, via the `card.aux.serial` API with the `mode` set to `notify,accel`, and `duration` set to 1000. This configures the Notecard to send accelerometer readings over the `TX/RX` serial connection once per second.
+
+When the app is first powered on, the normal orientation of the Notecard is determined from 10 seconds of accelerometer readings. While the Notecard (and Vending Machine) is stationary, the `x`, `y`, `z` readings from the accelerometer constitute a 3d vector that is the acceleration due to gravity, i.e. straight downwards. Depending upon the orientation of the Notecard inside the Vending Machine, the vector could be straight down, straight up, to the left, or any other direction that corresponds to the force of gravity relative to the Notecard.
+
+Once the downward vector has been determined, the app continues to receive accelerometer readings, and uses these to compute the current downward vector. The angle between the current downward vector and the original downward vector is computed to determine the current tilt of the vending machine, which is used to fire or clear the excessive tilt alert.
+
+#### Power Failure Alerts
+
+A power failure alert is sent when USB power fails and the app is running from the LiPo battery.
+
+```json
+{ "active": true, "text": "Power failure" }
+```
+
+On power failure, the app waits `DEFAULT_WAIT_BEFORE_SLEEP_MS` in case it's just a brief power glitch. Should power resume during that time, the alert is cleared and the app continues operating normally. Otherwise, the Notecard and Host sleep for 5 minute intervals (`DEFAULT_SLEEP_PERIOD_MS`), or until power is restored and the alert is cleared.
+
+```json
+{ "active": false, "text": "Power restored" }
+```
+
+#### Battery Low Alerts
+
+When running from battery, a battery low alert is sent when the battery voltage falls below
+`BATTERY_LOW_VOLTAGE` volts. (Default setting is 3.2v.)
```

Comment:
Probably can remove this newline, since we aren't breaking up paragraphs like that elsewhere.